shm/player.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "ship.hpp"
+#include "cargo.hpp"
```

Comment:
You can make a forward declaration and include both headers in .cpp file. This causes compilation faster because any changes in both files will not trigger rebuild this one.